app/models/user.rb

```diff
@@ -22,12 +28,38 @@ class User < RedisStore
   # Add additional MVI attributes
   alias redis_key uuid
 
+  # mvi 'golden record' data
+  attribute :mvi
+
   validates :uuid, presence: true
   validates :email, presence: true
 
+  after_initialize :fetch_mvi_data
+
   def self.sample_claimant
     attrs = JSON.load(ENV['EVSS_SAMPLE_CLAIMANT_USER'])
     attrs[:last_signed_in] = Time.now.utc
     User.new attrs
   end
+
+  def fetch_mvi_data
+    given_names = [first_name]
+    given_names.push middle_name unless middle_name.nil?
+    message = MVI::Messages::FindCandidateMessage.new(
+      given_names,
+      last_name,
+      dob,
+      ssn,
+      gender
+    )
+    if message.valid?
+      response = MVI_SERVICE.find_candidate(message)
+      update(mvi: response)
+    else
+      errors = message.errors.full_messages.join(', ')
+      Rails.logger.warn "MVI user data not retrieved: invalid message: #{errors}"
+    end
+  rescue MVI::ServiceError => e
```

Comment:
@robbiethegeek should we cross that bridge when get there maybe? As of right now I think all of the services for which the API would get invoked requires MVI data, raising a 400 level error, 401 or 400 here would seem to make a lot of sense. 

Later we could have a mechanism for determining whether or not the user who is signed in is a non-veteran and then NOT call fetch_mvi_data but I think fetch_mvi_data should be more deterministic for when it is invoked and not just be a nil value.